src/hooks/useRowSelection.ts

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext } from 'react';
+import type { SelectRowEvent } from '../types';
+
+const RowSelectionContext = createContext<boolean | undefined>(undefined);
+
+export const RowSelectionProvider = RowSelectionContext.Provider;
+
+const RowSelectionChangeContext =
+  createContext<((selectRowEvent: SelectRowEvent) => void) | undefined>(undefined);
+
+export const RowSelectionChangeProvider = RowSelectionChangeContext.Provider;
+
+export function useRowSelection(): readonly [boolean, (selectRowEvent: SelectRowEvent) => void] {
```

Comment:
FWIW the tuple is created on each call, and we don't care about it being mutable or immutable, so we can remove `readonly` here.
```suggestion
export function useRowSelection(): [boolean, (selectRowEvent: SelectRowEvent) => void] {
```